airgap/v2/cmd/client/list/list_test.go

```diff
@@ -0,0 +1,455 @@
+	"github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/cmd/client/util"
+	server "github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/cmd/server/start"
+	"github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/pkg/database"
+	"github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/pkg/models"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+const bufSize = 1024 * 1024
+
+var lis *bufconn.Listener
+var db database.Database
+var dbName = "client.db"
+
+func bufDialer(context.Context, string) (net.Conn, error) {
+	return lis.Dial()
+}
+
+func runSetup() {
+	//Initialize the mock connection and server
+	lis = bufconn.Listen(bufSize)
+	s := grpc.NewServer()
+	bs := server.BaseServer{}
+	mockSenderServer := server.FileSenderServer{}
+	mockRetreiverServer := server.FileRetreiverServer{}
+
+	//Initialize logger
+	zapLog, err := zap.NewDevelopment()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize zapr, due to error: %v", err))
+	}
+	bs.Log = zapr.NewLogger(zapLog)
+
+	//Create Sqlite Database
+	gormDb, err := gorm.Open(sqlite.Open(dbName), &gorm.Config{})
+	if err != nil {
+		logger.Fatalf("Error during creation of Database")
+	}
+	db.DB = gormDb
+	db.Log = bs.Log
+
+	//Create tables
+	err = db.DB.AutoMigrate(&models.FileMetadata{}, &models.File{}, &models.Metadata{})
+	if err != nil {
+		logger.Fatalf("Error during creation of Models: %v", err)
+	}
+
+	bs.FileStore = &db
+	mockSenderServer.B = bs
+	mockRetreiverServer.B = bs
+	filesender.RegisterFileSenderServer(s, &mockSenderServer)
+	fileretreiver.RegisterFileRetreiverServer(s, &mockRetreiverServer)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			if err.Error() != "closed" { //When lis of type (*bufconn.Listener) is closed, server doesn't have to panic.
+				panic(err)
+			}
+		} else {
+			logger.Printf("Mock server started")
+		}
+	}()
+}
+
+func createClient() *grpc.ClientConn {
+	ctx := context.Background()
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	if err != nil {
+		logger.Fatalf("failed to dial bufnet: %v", err)
+	}
+
+	return conn
+}
+
+func shutdown(conn *grpc.ClientConn) {
+	sqlDB, err := db.DB.DB()
+	if err != nil {
+		logger.Fatalf("Error: Couldn't close Database: %v", err)
+	}
+	sqlDB.Close()
+	conn.Close()
+	os.Remove(dbName)
+	lis.Close()
+}
+
+func TestList(t *testing.T) {
+	//Initialize the server
+	runSetup()
+	//Initialize connection
+	conn := createClient()
+	//Shutdown resources
+	defer shutdown(conn)
+
+	//Populate dataset for testing
+	populateDataset(conn, t)
+	listFilMetaDataCLient := fileretreiver.NewFileRetreiverClient(conn)
+	od, _ := os.Getwd()
+	tests := []struct {
+		name   string
+		lc     *Listconfig
+		res    []string
+		errMsg string
+	}{
+		{
+			name: "fetch one of the files just based on metadata and store to csv",
```

Comment:
Let's rename this to "fetch one of the files based on metadata and store to csv"